spec/lib/checklist_syntax_migrator_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ChecklistSyntaxMigrator do
+  before do
+    SiteSetting.allow_uncategorized_topics = true
+  end
+  let(:topic) { Fabricate(:topic) }
+  let(:post_args) do
+    { user: topic.user, topic: topic }
+  end
+
+  def post_with_body(body)
+    args = post_args.merge(raw: body)
+    Fabricate.build(:post, args)
+  end
+
+  it "replaces instances of the old checkbox instance, with the new syntax" do
+    body = "[-] 1\n[_] 2\n[*] 3\n[\\*] 4\n[\*] 5\n[*\] 6"
+    post = post_with_body(body)
+
+    ChecklistSyntaxMigrator.new(post).update_syntax!
+
+    expected = "[x] 1\n[x] 2\n[x] 3\n[x] 4\n[x] 5\n[x] 6"
+    expect(post.reload.raw).to eq(expected)
+  end
+
+  it "does not replace if more than 3 spaces are before a checkbox" do
+    body = "    [\*]\n      [-]"
+    post = post_with_body(body)
+
+    ChecklistSyntaxMigrator.new(post).update_syntax!
+    expect(post.reload.raw).to eq(body)
+  end
+
+  it "does not replace checkboxes after text" do
```

Comment:
Are checkboxes after text not converted to avoid false-positive migrations? Checkboxes in lists are a common case:

```
* [x] task1

- [ ] something

1. [ ] todo
```

Reply:
Yes, they are not converted to avoid false positives. This was Jeff's request

Reply:
I wonder how much value we'll get from this limited migration. 🤔

*(a couple minutes later)*

Okay, so I ran these on a certain *dev*elopmental forum:

```
Post.where("raw SIMILAR TO '%\\n\\s{0,3}\\[\\\\?\\*\\]%'").count
# => 125

Post.where("raw SIMILAR TO '%\s{0,3}([*-]|\\d+\\.)\\s\\[\\\\?\\*\\]%'").count
# => 146
```

So we're going to hit a bit less than the half of common cases. Good enough-ish? 😃

Reply:
Yeah, sounds good. Very standard posts that are primary checkboxes will be converted which is the main goal